integrations/phishing_analyzers/analyzers/driver_wrapper.py

```diff
@@ -0,0 +1,144 @@
+import functools
+import logging
+import os
+from enum import Enum
+from typing import Iterator
+
+from selenium.common import WebDriverException
+from seleniumwire.request import Request
+from seleniumwire.webdriver import ChromeOptions, Remote
+
+LOG_NAME = "driver_wrapper"
+
+# get flask-shell2http logger instance
+logger = logging.getLogger(LOG_NAME)
+# logger config
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+log_level = os.getenv("LOG_LEVEL", logging.INFO)
+log_path = os.getenv("LOG_PATH", f"/var/log/intel_owl/{LOG_NAME}")
+# create new file handlers, files are created if doesn't already exists
+fh = logging.FileHandler(f"{log_path}/{LOG_NAME}.log")
+fh.setFormatter(formatter)
+fh.setLevel(log_level)
+fh_err = logging.FileHandler(f"{log_path}/{LOG_NAME}_errors.log")
+fh_err.setFormatter(formatter)
+fh_err.setLevel(logging.ERROR)
+# add the handlers to the logger
+logger.addHandler(fh)
+logger.addHandler(fh_err)
+logger.setLevel(log_level)
+
+
+class DriverWrapperActions(Enum):
+    ACTION_NAVIGATE = "navigate"
+    ACTION_PAGE_SOURCE = "page_source"
+    ACTION_CURRENT_URL = "current_url"
+    ACTION_BASE64_SCREENSHOT = "base64_screenshot"
+
+
+def driver_exception_handler(action: DriverWrapperActions = ""):
```

Comment:
I think that the structure DriverWrapperActions you created on top of the decorator is not necessary and overcomplicates the code. Correct me if I am wrong, but that seems necessary only for changing a very little part of the error log. In such case, if the `url` does not exist, you could simply logging it as empty.

Reply:
Ok I removed the structure.
I would like to keep the if on `url` because it can be misleading reading `None` in logs since this value is populated only when action is `"navigate"`.